File: _정답코드/2주차/07_최대_적재하중/solution.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<int> ropes) {

    sort(ropes.begin(), ropes.end(), greater<int>());

	int answer = -1;
    for(int i=0; i<ropes.size(); i++){
        answer = max(answer, (i+1) * ropes[i]);
    }
	return answer;
}

File: _정답코드/2주차/07_최대_적재하중/Solution.java
import java.util.*;

public class Solution {
	public int solution(int[] ropes) {

        Integer[] boxedRopes = Arrays.stream(ropes)
            .boxed()
            .toArray(Integer[]::new);
        Arrays.sort(boxedRopes, Collections.reverseOrder());

        int answer = -1;
        for(int i = 0; i<boxedRopes.length; i++){
            answer = Math.max(answer, (i + 1) * boxedRopes[i]);
        }

        return answer;
	}
}

